frontend/packages/console-plugin-sdk/src/typings/details.ts

```diff
@@ -0,0 +1,23 @@
+import { K8sKind } from '@console/internal/module/k8s';
+import { PageComponentProps } from '@console/internal/components/utils/horizontal-nav';
```

Comment:
@alecmerdler This is effectively about extending the common `HorizontalNav` component which is used on many places. If I read the code correctly, this covers more than just resource details pages.

I'd suggest to give this extension a more accurate name. For example, `HorizontalNavTab` as extension type name and `HorizontalNav/Tab` as `type` property value.

In general, the `type` value should reflect the context of the extension. For example, `Dashboards/Tab` means a tab on dashboards page, and `Dashboards/Overview/Inventory/Item` means new item in inventory which is part of overview on dashboards page.